### PR TITLE
[Fix #6057] Change auto-gen-config return value

### LINKED ISSUE #6057: rubocop --auto-gen-conf exits with 1
When rubocop find offenses it exits with 1, which in general is expected. But when running `rubocop --auto-gen-conf ` I would expect that it returns 1 only if something went wrong and 0 in other case, as that there are offenses is expected and doesn't mean that the execution of the command went wrong. :confused:

### PR DESCRIPTION
**Pull request by MagedMilad**
Fixes #6057 

## Summary
This PR fixes the return value of `rubocop --auto-gen-conf` to return 0 when the todo file is successfully created, without taking into account if there are offenses
```diff
diff --git a/lib/rubocop/cli.rb b/lib/rubocop/cli.rb
index 3844f01..018f58d 100644
--- a/lib/rubocop/cli.rb
+++ b/lib/rubocop/cli.rb
@@ -159,7 +159,9 @@ module RuboCop
       display_error_summary(runner.errors)
       maybe_print_corrected_source
 
-      if all_passed && !runner.aborting? && runner.errors.empty?
+      all_pass_or_excluded = all_passed || @options[:auto_gen_config]
+
+      if all_pass_or_excluded && !runner.aborting? && runner.errors.empty?
         STATUS_SUCCESS
       else
         STATUS_OFFENSES
```
-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [X] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [X] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 * [#6124](https://github.com/rubocop-hq/rubocop/issues/6124): Fix `Style/IfUnlessModifier` cop for disabled `Layout/Tab` cop when there is no `IndentationWidth` config. ([@AlexWayfer][])
 * [#6133](https://github.com/rubocop-hq/rubocop/pull/6133): Fix `AllowURI` option of `Metrics/LineLength` cop for files with tabs indentation. ([@AlexWayfer][])
 * [#6164](https://github.com/rubocop-hq/rubocop/issues/6164): Fix incorrect autocorrect for `Style/UnneededCondition` when using operator method higher precedence than `||`. ([@koic][])
+* [#6057](https://github.com/rubocop-hq/rubocop/issues/6057): Fix return 0 when running `rubocop --auto-gen-conf` if the todo file is successfully created even if there are offenses. ([@MagedMilad][])
```
Comment (Ana06) [inline CHANGELOG.md]: I am not sure if this is a bug or more a change :thinking:

File: lib/rubocop/cli.rb [hunk 2]
```diff
@@ -159,7 +159,9 @@ def execute_runner(paths)
       display_error_summary(runner.errors)
       maybe_print_corrected_source
 
-      if all_passed && !runner.aborting? && runner.errors.empty?
+      pass_condition = all_passed || @options[:auto_gen_config]
```
Comment (Ana06) [inline lib/rubocop/cli.rb]: I would give this a different name, which tells you this is a boolean and is more explicit about what is actually doing, what about `all_pass_or_excluded`?

Comment (MagedMilad) [reply lib/rubocop/cli.rb]: ok `all_pass_or_excluded` seems better

**Comment by MagedMilad** [comment]
@Ana06 thank you for your review, just considered your suggestion

**Comment by bbatsov** [comment]
Looks good to me! Thanks!

**Pull request merged**